build/crds/build-crds.sh

```diff
@@ -54,8 +54,6 @@ copy_ob_obc_crds() {
 generating_crds_v1() {
   echo "Generating ceph crds"
   "$CONTROLLER_GEN_BIN_PATH" "$CRD_OPTIONS" paths="./pkg/apis/ceph.rook.io/v1" output:crd:artifacts:config="$OLM_CATALOG_DIR"
-  # the csv upgrade is failing on the volumeClaimTemplate.metadata.annotations.crushDeviceClass unless we preserve the annotations as an unknown field
-  $YQ_BIN_PATH eval --inplace '.spec.versions[0].schema.openAPIV3Schema.properties.spec.properties.storage.properties.storageClassDeviceSets.items.properties.volumeClaimTemplates.items.properties.metadata.properties.annotations.x-kubernetes-preserve-unknown-fields = true' "${OLM_CATALOG_DIR}"/ceph.rook.io_cephclusters.yaml
```

Comment:
should I keep these changes so that we have less conflicts with downstream when backporting?

The above changes impacts the crds.yaml and if there are changes in upstream and downstream crds.yaml we'll face conflicts whenever the upstream crds.yaml is updated

Reply:
If it affects crds.yaml, sounds good to keep it. How does it affect the crds.yaml? It looks like this just evaluates on `ceph.rook.io_cephclusters.yaml`?

Reply:
crds.yaml contains the `ceph.rook.io_cephclusters.yaml` crd, basically `ceph.rook.io_cephclusters.yaml` is a subset of crds.yaml and `   x-kubernetes-preserve-unknown-fields: true` this is removed

Reply:
added this line